brain/morphology.h

```diff
@@ -0,0 +1,132 @@
+
+/* Copyright (c) 2013-2015, EPFL/Blue Brain Project
+ *                          Juan Hernando <jhernando@fi.upm.es>
+ *
+ * This file is part of Brion <https://github.com/BlueBrain/Brion>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef BRAIN_MORPHOLOGY
+#define BRAIN_MORPHOLOGY
+
+#include <brain/api.h>
+#include <brain/types.h>
+
+#include <boost/noncopyable.hpp>
+
+namespace brain
+{
+
+/**
+ * Wrapper around brion::Morphology with higher level functions.
+ *
+ * This class provides methods to facilitate some queries about morphologies
+ * in the context of circuits.
+ * Morphologies can be loaded with a transformation applied to its points,
+ * which is useful for operating in global circuit coordinates.
+ * The transformation is applied at construction so it cannot be modified or
+ * reverted.
+ *
+ * Access to the raw data fields is still provided by getter functions.
+ *
+ * @version unstable
+ */
+class Morphology : public boost::noncopyable
+{
+public:
+    /**
+     * Create a morphology from a URI, load all the data and transform
+     * the points.
+     * @param transform the transformation matrix to apply to the points.
+     * @param source URI of the morphology data source.
+     * @param stage the repair stage to load.
+     */
+    BRAIN_API Morphology( const URI& source, const Matrix4f& transform,
+                          brion::MorphologyRepairStage stage =
```

Comment:
OK, very interesting to know! I see you updated already; I think it's a good to keep this high-level API as simple as possible. We can add the option later if the need arises, but it's hard to remove it once it's already there.